### DISCUSSION TITLE
How to add gradient or hessian matrix for fmincon in yalmip

### DISCUSSION THREAD

**Discussion by NickJay2** [yalmip/YALMIP#1380] [Optimization modelling]
I have a very complicated objective function, which results that the optimization not completed even after 10000 iterations. Actually, I can give the analytical solution of hessian matrix. Will accurate hessian matrix accelerate the optimization? If yes, how can I write the code to add analytical solution of the hessian matrix. Part of my cods is as below.
`     qx=sdpvar(N,1);
     qy=sdpvar(N,1);
     Constraints=[qx(1)==qx_I,qy(1)==qy_I,qx(N)==qx_F,qy(N)==qy_F];
     for n=1:N-1
         Constraints=[Constraints,((qx(n+1)-qx(n))^2+(qy(n+1)-qy(n))^2)<=(V_max*delta_t)^2];
     end
     obj=0;
     for n=1:N
         obj=obj+sum(R_klb(a(n,:),b(n,:),gama0,H,C1,C2,B1,B2,wx,wy,qx(n),qy(n),qx_r(n),qy_r(n)).*...
             exp(-alpha*R_klb(a(n,:),b(n,:),gama0,H,C1,C2,B1,B2,wx,wy,qx(n),qy(n),qx_r(n),qy_r(n))))/...
             sum(exp(-alpha*R_klb(a(n,:),b(n,:),gama0,H,C1,C2,B1,B2,wx,wy,qx(n),qy(n),qx_r(n),qy_r(n))));
     end
     Objective=-obj/N;
     ops = sdpsettings('fmincon.maxfunevals',10000,'fmincon.MaxIter',3000);
     sol=optimize(Constraints,Objective,ops); `

**Comment by johanlofberg** [comment]
Gradients are automatically computed and supplied to fmincon. Hessians are not supported so fmincon will use quasi-newton if it uses hessian

**Reply by NickJay2** [reply]
So, is there any method I can take to accelerate it? I may need to repeat the optimization about 1000 times (over 1000*10000 iterations), which is a quite hard task.

**Reply by johanlofberg** [reply]
Well, you can always try tweaking algorithmic settings in fmincon such as switching algorithm, but most likely you have to come up with a better way to represent the objective. If you can compute gradients and hessians, then it sounds like a pretty simple compact model and you might be better off simply interfacing fmincon directly, avoiding any kind of extra variables etc that are introduced by yalmip
